Mathlib/Topology/Separation/LinearUpperLowerSetTopology.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2025 Miyahara Kō. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miyahara Kō
+-/
+
+import Mathlib.Topology.Order.UpperLowerSetTopology
+import Mathlib.Topology.Separation.Regular
+
+/-!
+# Linear upper or lower sets topologies are completely normal
+-/
+
+open Set Topology.IsUpperSet
+
+instance (priority := low) {α : Type*}
+    [TopologicalSpace α] [LinearOrder α] [Topology.IsUpperSet α] :
+    CompletelyNormalSpace α where
+  completely_normal s t hcst hsct := by
+    obtain (rfl | ⟨a, ha⟩) := s.eq_empty_or_nonempty
+    case inl => simp
+    obtain (rfl | ⟨b, hb⟩) := t.eq_empty_or_nonempty
+    case inl => simp
+    exfalso
+    -- issue #27058
+    fail_if_success grw [← singleton_subset_iff.mpr ha, ← singleton_subset_iff.mpr hb] at hcst hsct
```

Comment:
I'm assuming you checked this with current `master` (which includes the PR mentioned at #27058) and it still doesn't work? Could you post an update on that issue, mentioning this PR?